nengo_dl/transform_builders.py

```diff
@@ -165,13 +191,38 @@ def build_step(self, signals):
             W = tf.transpose(W, perm=self.perm_w)
             W = tf.reshape(W, self.reshape_w)
 
-        Y = tf.nn.convolution(
-            input=X,
-            filters=W,
-            strides=self.conv.strides,
-            data_format=self.fmt,
-            padding=self.conv.padding.upper(),
-        )
+        if self.transpose:
+            output_space = list(self.conv.output_shape.spatial_shape)
+            output_filters = sum(op.conv.n_filters for op in self.ops)
+
+            output_shape = [X.shape[0]] + (
+                output_space + [output_filters]
+                if self.conv.channels_last
+                else [output_filters] + output_space
+            )
+            if self.perm_x is not None:
+                output_shape = [output_shape[i] for i in self.perm_x]
+
+            # swap channels, because conv_transpose order is for forward weights
+            dims = self.conv.dimensions
+            filters = tf.transpose(W, perm=list(range(dims)) + [dims + 1, dims])
```

Comment:
Similar to above comment about the `tf.constant`, I think all of this could be precomputed in `build_pre` because it doesn't depend on `X` (except for the batch size, which can just be specified as `-1` I'd expect), which would be more efficient.

Reply:
Replacing `X.shape[0]` with `-1` seems to work (at least for the test suite).